nx/lib/nx/defn/kernel.ex

```diff
@@ -522,6 +522,34 @@ defmodule Nx.Defn.Kernel do
 
   def left and right, do: Nx.logical_and(left, right)
 
+  @doc """
+  Element-wise logical AND operation.
+
+  Zero is considered false, all other numbers
+  are considered true.
+
+  It delegates to `Nx.logical_and/2` (supports broadcasting).
+
+  ## Examples
+
+      defn and_or(a, b) do
+        {a && b, a || b}
+      end
+
+  """
+  def left && right when Kernel.or(is_boolean(left), is_boolean(right)) do
+    raise ArgumentError,
+          "boolean value passed to Nx.Defn.Kernel.and/2, " <>
+            "values passed to Nx.Defn.Kernel.and/2 must be " <>
+            "tensors or numbers, consider using 1 for true " <>
+            "and 0 for false as an alternative"
+  end
+
+  def left && right when Kernel.and(is_number(left), is_number(right)),
+    do: logical_and(left, right)
+
+  def left && right, do: Nx.logical_and(left, right)
```

Comment:
I'm not sure if this is actually the implementation we want. This was discussed because with `&&` and `||` we can mostly avoid going into `transform` just to extract opts.

For example: `eps = opts[:eps] || 1.0e-5`, So I think it's more in line with adding something closer to `if` or `cond`

Reply:
ok it makes sense, in this simple implementation i just reflected the behavior of the `and` and `or` operator.

I can adjust.